pyadjoint/reduced_functional.py

```diff
@@ -206,10 +206,13 @@ def __call__(self, values):
         blocks = self.tape.get_blocks()
         with self.marked_controls():
             with stop_annotating():
-                for i in self.tape._bar("Evaluating functional").iter(
-                    range(len(blocks))
-                ):
-                    blocks[i].recompute()
+                if self.tape._checkpoint_manager is not None:
```

Comment:
```suggestion
                if self.tape._checkpoint_manager:
```

Reply:
I don't think that this is always the right thing to do. From [PEP8](https://peps.python.org/pep-0008/#programming-recommendations):
> Also, beware of writing if x when you really mean if x is not None – e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!